purchase_ext_sst/models/purchase_order.py

```diff
@@ -51,6 +51,16 @@ def onchange_shop_id(self):
             if self.purchased_by_id and self.purchased_by_id.shop_id and \
                     self.purchased_by_id.shop_id != self.shop_id:
                 self.purchased_by_id = False
+            # Update picking_type_id
+            picking_type_id = self.env['stock.picking.type'].search([
+                ('code', '=', 'incoming'),
+                ('warehouse_id', '=', self.shop_id.id),
+                ('warehouse_id.company_id', 'in', [self.env.context.get(
+                    'company_id', self.env.user.company_id.id),
+                    False]),
```

Comment:
@TimLai125 I'm late to comment, but isn't it redundant to include company_id in the domain here?

Reply:
Maybe it is kind of redundant, but I grepped this domain from here: 
https://github.com/odoo/odoo/blob/b8f8012af8e9a7b9974a6578133e81ea890b0f23/addons/mrp/models/mrp_production.py#L20-L25

Reply:
It would be valid to have the company_id in the domain only in case `shop_id` somehow belongs to the wrong company, which shouldn't happen under normal circumstances. So, I'd prefer to take this off. At the same time I guess it doesn't cause any harm, so we can leave it as it is as well.

Reply:
ok, I got it. I will remove this part later on this evening.